src/core/Metadata.cpp

```diff
@@ -265,6 +266,19 @@ int Metadata::historyMaxSize() const
     return m_data.historyMaxSize;
 }
 
+int Metadata::autosaveDelayMin() const
```

Comment:
I wanted to recommend to write a unit test for this new method but figured out there are no unit tests for `core` classes at all. Interesting why? We can start a new tradition to write unit tests for `core` classes :)

Reply:
In the coverage report it seems i use all functions i added in Metadata, is that reasonable or should there be a test the specificly calls `Metadata::autosaveDelayMin()`? Where should i include such test?

Reply:
Coverage report does not prove the tests are correct and covers all possible use cases.
For the `Metadata::autosaveDelayMin()` I see the next test cases:

- The `customData` does not have `KPXC_autosaveDelayMin` key.
- `KPXC_autosaveDelayMin` value is not defined (null).
- `KPXC_autosaveDelayMin` value is defined but not a number, say empty string or some text.
- `KPXC_autosaveDelayMin` value is defined correctly.

If all these cases are covered by GUI tests, fine. But I am talking about dedicated tests: `test/core/TestMetadata` class. This is the only correct test class for your changes. As I mentioned above we don't have the `test/core` folder at all. If so, it is good time to start writing unit tests for `core` classes. But I don't insist, it is up to you. Let's ask @droidmonkey for advise.

Reply:
I could attempt adding this, @droidmonkey let me know if that seems right

Reply:
* The `customData` does not have `KPXC_autosaveDelayMin` key.
Covered by opening the database settings widget.
 * `KPXC_autosaveDelayMin` value is not defined (null).
Covered by opening the database settings widget.
 * `KPXC_autosaveDelayMin` value is defined but not a number, say empty string or some text.
This can't be done from gui, so i think it can only fit in a new testing file for the Metadata class, which i doubt should be added as part of this PR
 * `KPXC_autosaveDelayMin` value is defined correctly.
This is covered in the added tests